tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.h

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_COMMON_RUNTIME_GPU_GPU_CUDAMALLOCASYNC_ALLOCATOR_H_
+#define TENSORFLOW_CORE_COMMON_RUNTIME_GPU_GPU_CUDAMALLOCASYNC_ALLOCATOR_H_
+
+#include <memory>
+
+#include "third_party/gpus/cuda/include/cuda.h"
+#include "third_party/gpus/cuda/include/cuda_runtime_api.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "tensorflow/core/common_runtime/gpu/gpu_id.h"
+#include "tensorflow/core/framework/allocator.h"
+#include "tensorflow/core/platform/macros.h"
+#include "tensorflow/core/platform/stream_executor.h"
+#include "tensorflow/core/platform/types.h"
+
+namespace tensorflow {
+
+// An allocator that wraps cudaMallocAsync. It has less fragmentation
+// issues then the BFC memory allocator.  The compute-sanitizer tool
+// helps to detect OOB memory error of cudaMallocAsync. Use the
+// environment variable TF_GPU_ALLOCATOR=cuda_malloc_async to enable
+// it.
+//
+// It needs CUDA 11.2+. When using a container, this only needs the
+// container driver to be 11.2. It has a WAR again a driver bug in
+// multi-GPU with CUDA 11.2. The WAR creates an extra context on GPU 0.
+//
+// We configure cudaMallocAsync to grow when more memory is needed
+// instead of preallocating everything up front.  But it never releases
+// to other process the GPU memory.  So no other process will "steal"
+// the GPU memory already used by the current process. This is to
+// prevent crashes of long running jobs.  Use 'reserve_memory=true' if
+// you want to preallocate the memory.
+class GPUcudaMallocAsyncAllocator : public Allocator {
```

Comment:
style: Abbreviations should be treated like words, so this type should be GpuCudaMallocAsyncAllocator

see: https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
I think it would be better to check this one in correctly the first time (since this is a trivial change) and we can carefully change the other one separately

Reply:
done